ClientFactory/PluginClientFactory.php

```diff
@@ -2,6 +2,8 @@
 
 namespace Http\HttplugBundle\ClientFactory;
 
+@trigger_error('The '.__NAMESPACE__.'\PluginClientFactory class is deprecated since version 1.8 and will be removed in 2.0.', E_USER_DEPRECATED);
```

Comment:
we could also add a `@deprecated` annotation in the class docblock to make IDEs show that there is a problem when using this class.